example/chessground/lib/round_screen.dart

```diff
@@ -0,0 +1,765 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:ble_backend/ble_connector.dart';
+import 'package:ble_backend/ble_peripheral.dart';
+import 'package:ble_backend_screens/ui/ui_consts.dart';
+import 'package:ble_chess_example/options_screen.dart';
+import 'package:ble_chess_peripheral_driver/ble_chess_peripheral_driver.dart';
+import 'package:ble_chess_peripheral_driver/chess_peripheral_driver.dart';
+import 'package:chessground/chessground.dart';
+import 'package:dartchess/dartchess.dart';
+import 'package:fast_immutable_collections/fast_immutable_collections.dart';
+import 'package:flutter/material.dart';
+import 'package:fluttertoast/fluttertoast.dart';
+import 'peripheral_fen.dart';
+
+class RoundScreen extends StatefulWidget {
+  RoundScreen({
+    required this.bleConnector,
+    required this.blePeripheral,
+    super.key,
+  });
+
+  final BleConnector bleConnector;
+  final BlePeripheral blePeripheral;
+
+  @override
+  State<RoundScreen> createState() => RoundScreenState();
+}
+
+enum PlayMode {
+  bot,
+  free,
+}
+
+enum Ack {
+  yes,
+  no,
+}
+
+class RoundScreenState extends State<RoundScreen> {
+  StreamSubscription? _subscription;
+  Peripheral peripheral = DummyPeripheral();
+  bool isAutocompleteOngoing = false;
+  bool isOfferOngoing = false;
+  Position position = Chess.initial;
+  Side orientation = Side.white;
+  String fen = kInitialBoardFEN;
+  NormalMove? lastMove;
+  NormalMove? promotionMove;
+  NormalMove? premove;
+  ValidMoves validMoves = IMap(const {});
+  PlayMode playMode = PlayMode.free;
+  Position? lastPos;
+
+  BlePeripheral get blePeripheral => widget.blePeripheral;
+  BleConnector get bleConnector => widget.bleConnector;
+
+  Future<void> _beginNewRound() async {
+    setState(() {
+      position = Chess.initial;
+      fen = position.fen;
+      validMoves = makeLegalMoves(position);
+      lastMove = null;
+      lastPos = null;
+    });
+    playMode = await _showChoicesPicker<PlayMode>(
+      context: context,
+      title: 'Select play mode',
+      choices: PlayMode.values,
+      defaultValue: playMode,
+    );
+    await peripheral.handleBegin(
+      fen: fen,
+      variant: Variants.standard,
+      side: playMode == PlayMode.bot ? Sides.white : Sides.both,
+      lastMove: lastMove?.uci,
+      check: _getCheck(),
+    );
+  }
+
+  String? _getCheck() {
+    final king = position.board.kingOf(position.turn);
+    return king != null && position.checkers.isNotEmpty ? king.name : null;
```

Comment:
is it correct condition?
checkers list cen be white and black pawns...right?

Reply:
I just copied from `position.isCheck` that return `bool` instead `Square`